src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -165,19 +232,43 @@ function(mode, isCurve, isBezier,isQuadratic, isContour, shapeKind){
   else {
     gl.enable(gl.BLEND);
     gl.drawArrays(this.immediateMode.shapeMode, 0,
-      this.immediateMode.vertexPositions.length / 3);
+      this.immediateMode.vertexPositions.length);
   }
-  //clear out our vertexPositions & colors arrays
-  //after rendering
-  this.immediateMode.vertexPositions.length = 0;
-  this.immediateMode.vertexColors.length = 0;
-  this.immediateMode.uvCoords.length = 0;
-  this.isImmediateDrawing = false;
+  // todo / optimizations? leave bound until another shader is set?
+  fillShader.unbindShader();
+};
 
+p5.RendererGL.prototype._drawStrokeImmediateMode = function() {
+  var gl = this.GL;
+  var strokeShader = this.curStrokeShader;
+  strokeShader.bindShader();
+  this._bindBuffer(this.immediateMode.lineVertexBuffer, gl.ARRAY_BUFFER,
+    flatten(this.immediateMode.lineVertices), Float32Array, gl.STATIC_DRAW);
+  strokeShader.enableAttrib(strokeShader.attributes.aPosition.location,
+    3, gl.FLOAT, false, 0, 0);
+  this._bindBuffer(this.immediateMode.lineNormalBuffer, gl.ARRAY_BUFFER,
+    flatten(this.immediateMode.lineNormals), Float32Array, gl.STATIC_DRAW);
+  strokeShader.enableAttrib(strokeShader.attributes.aDirection.location,
+    4, gl.FLOAT, false, 0, 0);
+  gl.drawArrays(gl.TRIANGLES, 0,
+      this.immediateMode.lineVertices.length);
   // todo / optimizations? leave bound until another shader is set?
-  shader.unbindShader();
-  return this;
+  strokeShader.unbindShader();
 };
 
+function flatten(arr){
+  if (arr.length>0){
+    return ([].concat.apply([], arr));
+  } else {
+    return [];
+  }
+}
+
+function vToNArray(arr){
+  return flatten(arr.map(function(item){
+    return [item.x, item.y, item.z];
+  }));
+}
```

Comment:
Is the arr here a p5.Vector? Can we use normal JS arrays instead? 
then I think you could just call flatten on that array and skip this step.

Reply:
Not sure how this can be done cleanly. A number of the prototypes that come along with the p5.Vector class are used. The problem mainly comes with abstracting `edgesToVertices` such that both immediate and retained mode use the same function. In doing so I have to assume that the Immediate Mode `vertexPositions` are also p5.Vector object because the retained mode positions need to be because there is extensive use of p5.Vector functions in calculating normals (among other things)